crypto/ocsp/ocsp_test.cc

```diff
@@ -199,22 +264,331 @@ static const uint8_t ocsp_response_der[] = {
+                     const char *pem) {
+  bssl::UniquePtr<BIO> bio(BIO_new_mem_buf(pem, strlen(pem)));
+  if (!bio) {
+    return false;
+  }
+
+  char *name, *header;
+  uint8_t *data;
+  long data_len;
+  if (!PEM_read_bio(bio.get(), &name, &header, &data, &data_len)) {
+    fprintf(stderr, "failed to read PEM data.\n");
+    return false;
+  }
+  OPENSSL_free(name);
+  OPENSSL_free(header);
+
+  out->reset(data);
+  *out_len = data_len;
+  return true;
+}
+
+static bssl::UniquePtr<X509> LoadX509fromPEM(const char *pem) {
+  size_t data_len;
+  bssl::UniquePtr<uint8_t> data;
+  if(!PEMToDER(&data, &data_len, pem)){
+    return nullptr;
+  }
+
+  bssl::UniquePtr<CRYPTO_BUFFER> buf(
+      CRYPTO_BUFFER_new(data.get(), data_len, nullptr));
+  if(!buf){
+    return nullptr;
+  }
+
+  return bssl::UniquePtr<X509>(X509_parse_from_buffer(buf.get()));
+}
+
+static void ExtractBasicOCSP(bssl::Span<const uint8_t> der,
+                             int ocsp_status,
+                             bssl::UniquePtr<OCSP_BASICRESP> *basic_response){
   bssl::UniquePtr<OCSP_RESPONSE> ocsp_response;
-  bssl::UniquePtr<OCSP_BASICRESP> basic_response;
 
-  ocsp_response = LoadOCSP_RESPONSE(ocsp_response_der);
+  ocsp_response = LoadOCSP_RESPONSE(der);
   ASSERT_TRUE(ocsp_response);
 
-  int ocsp_status = OCSP_response_status(ocsp_response.get());
-  ASSERT_EQ(OCSP_RESPONSE_STATUS_SUCCESSFUL, ocsp_status);
+  int ret = OCSP_response_status(ocsp_response.get());
+  ASSERT_EQ(ocsp_status, ret);
+
+  *basic_response = bssl::UniquePtr<OCSP_BASICRESP>(OCSP_response_get1_basic(ocsp_response.get()));
+  ASSERT_TRUE(*basic_response);
+}
+
+TEST(OCSPTest, TestBasic) {
+  bssl::UniquePtr<OCSP_BASICRESP> basic_response;
+  ExtractBasicOCSP(ocsp_response_der, OCSP_RESPONSE_STATUS_SUCCESSFUL, &basic_response);
+  ASSERT_TRUE(basic_response);
+}
 
-  basic_response = bssl::UniquePtr<OCSP_BASICRESP>(OCSP_response_get1_basic(ocsp_response.get()));
+TEST(OCSPTest, TestGoodOCSP) {
+  bssl::UniquePtr<OCSP_BASICRESP> basic_response;
+  ExtractBasicOCSP(ocsp_response_der, OCSP_RESPONSE_STATUS_SUCCESSFUL, &basic_response);
   ASSERT_TRUE(basic_response);
+
+  // Set up trust store and certificate chain
+  bssl::UniquePtr<X509_STORE> trust_store(X509_STORE_new());
+  X509_STORE_add_cert(trust_store.get(),LoadX509fromPEM(ca_cert).get());
+  bssl::UniquePtr<STACK_OF(X509)> server_cert_chain = CertsToStack(
+      {LoadX509fromPEM(server_cert).get(),LoadX509fromPEM(ca_cert).get()});;
+
+  X509 *subject = sk_X509_value(server_cert_chain.get(), 0);
+  X509 *issuer = sk_X509_value(server_cert_chain.get(), 1);
+
+  // Expect basic verify here, but we skip step for now since functionality has
+  // not been implemented yet.
+
+  int status = 0;
+  int reason = 0;
+  // Convert issuer certificate to |OCSP_CERTID|
+  bssl::UniquePtr<OCSP_CERTID> cert_id = bssl::UniquePtr<OCSP_CERTID>(OCSP_cert_to_id(EVP_sha1(), subject, issuer));
```

Comment:
Considering the discussion about `sha1`. Maybe we should add a call with a `NULL` parameter to catch if the behaviour of `OCSP_cert_to_id` changes? Add a comment above the call to explain why it is added.

Reply:
I added a second note in the function description for OCSP in `ocsp.h` to clarify this behavior and another test to see if the OCSP parsing behavior is effected when setting a `NULL` parameter in `OCSP_cert_to_id`.